tests/kernel/workq/work_queue_api/src/main.c

```diff
@@ -129,9 +130,14 @@ static void tdelayed_work_submit(void *data)
 			zassert_true(k_delayed_work_submit(&delayed_work[i],
 							   TIMEOUT) == 0, NULL);
 		}
-		/**TESTPOINT: check remaining timeout after submit*/
-		zassert_true(k_delayed_work_remaining_get(&delayed_work[i]) >=
-			     TIMEOUT, NULL);
+
+		time_remaining = k_delayed_work_remaining_get(&delayed_work[i]);
+
+		/**TESTPOINT: check remaining timeout after submit */
+		zassert_true(
+		    time_remaining <= __ticks_to_ms(z_ms_to_ticks(TIMEOUT)) &&
+		    time_remaining >= __ticks_to_ms(z_ms_to_ticks(TIMEOUT) - 1),
```

Comment:
This seems dangerous.  You're making the valid boundary only 1ms wide.  Remember that qemu platforms like to "time warp" forward by ~1 host scheduling quantum (1ms on Fedora, where CONFIG_HZ=1000, but on Ubuntu it's 2.5ms!).  I'd be worried these bounds wouldn't work reliably in CI.

Reply:
As you predicted, the current patch fails on emulated targets.
I will raise the limit to 15ms (more than 1 default tick).

Reply:
I updated the code. I also taken under account _TICK_ALIGN. Please check.